src/Execution/DataLoader/RelationCountBatchLoader.php

```diff
@@ -39,23 +40,27 @@ public function __construct(
      */
     public function resolve(): array
     {
-        return $this->getParentModels()
-            ->loadCount([$this->relationName => $this->decorateBuilder])
-            ->all();
+        return $this->getParentModels()->all();
     }
 
     /**
-     * Get the parents from the keys that are present on the BatchLoader.
+     * Get the parent models from the keys that are present on the BatchLoader.
+     * Models are grouped by their fully qualified class name to prevent key
+     * collisions between different types of models.
      */
-    protected function getParentModels(): EloquentCollection
+    protected function getParentModels(): Collection
```

Comment:
What's in the collection? Please add a type hint such as `@return Collection<?>`